modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/editor/configuration/CommerceCKEditorConfigContributor.java

```diff
@@ -0,0 +1,203 @@
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.commerce.product.definitions.web.internal.editor.configuration;
+
+import com.liferay.ai.creator.openai.configuration.manager.AICreatorOpenAIConfigurationManager;
+import com.liferay.commerce.product.constants.CPPortletKeys;
+import com.liferay.document.library.kernel.processor.AudioProcessorUtil;
+import com.liferay.portal.kernel.editor.configuration.BaseEditorConfigContributor;
+import com.liferay.portal.kernel.editor.configuration.EditorConfigContributor;
+import com.liferay.portal.kernel.json.JSONArray;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.module.configuration.ConfigurationException;
+import com.liferay.portal.kernel.portlet.LiferayWindowState;
+import com.liferay.portal.kernel.portlet.RequestBackedPortletURLFactory;
+import com.liferay.portal.kernel.portlet.url.builder.PortletURLBuilder;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.Validator;
+
+import java.util.Map;
+
+import javax.portlet.PortletMode;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Andrea Sbarra
+ */
+@Component(
+	property = {
+		"editor.name=ckeditor",
+		"javax.portlet.name=" + CPPortletKeys.CP_DEFINITIONS
+	},
+	service = EditorConfigContributor.class
+)
+public class CommerceCKEditorConfigContributor
+	extends BaseEditorConfigContributor {
+
+	@Override
+	public void populateConfigJSONObject(
+		JSONObject jsonObject, Map<String, Object> inputEditorTaglibAttributes,
+		ThemeDisplay themeDisplay,
+		RequestBackedPortletURLFactory requestBackedPortletURLFactory) {
+
+		if (!_isAICreatorChatGPTGroupEnabled(
+				themeDisplay.getCompanyId(), themeDisplay.getScopeGroupId())) {
+
+			return;
+		}
+
+		String extraPlugins = (String)jsonObject.get("extraPlugins");
+
+		JSONArray toolbarSimpleJSONArray = _getToolbarSimpleJSONArray();
+
+		jsonObject.put(
+			"aiCreatorOpenAIURL",
+			() -> PortletURLBuilder.create(
+				requestBackedPortletURLFactory.createControlPanelRenderURL(
+					"com_liferay_ai_creator_openai_web_internal_portlet_" +
+						"AICreatorOpenAIPortlet",
+					themeDisplay.getScopeGroup(),
+					themeDisplay.getRefererGroupId(), 0)
+			).setMVCPath(
+				"/view.jsp"
+			).setPortletMode(
+				PortletMode.VIEW
+			).setWindowState(
+				LiferayWindowState.POP_UP
+			).buildString()
+		).put(
+			"aiCreatorPortletNamespace",
+			() -> _portal.getPortletNamespace(
+				"com_liferay_ai_creator_openai_web_internal_portlet_" +
+					"AICreatorOpenAIPortlet")
+		).put(
+			"extraPlugins", extraPlugins.concat(",aicreator")
+		).put(
+			"isAICreatorOpenAIAPIKey",
```

Comment:
is it correct that starts with **is**? I think it's a string

Reply:
I've copied pasted this from the webcontent editor, I think this is right